contracts/smart-account/test/upgrades/v1/BaseSmartAccountV1.sol

```diff
@@ -127,24 +131,22 @@ abstract contract BaseSmartAccount is IAccount, BaseSmartAccountErrorsV1 {
     }
 
     /**
-     * @dev Initialize the Smart Account with required states
-     * @param _owner Signatory of the Smart Account
-     * @param _handler Default fallback handler provided in Smart Account
-     * @notice devs need to make sure it is only callble once by initiazer or state check restrictions
-     */
-    function init(address _owner, address _handler) external virtual;
-
-    /**
-     * @dev Gnosis style transaction with optional repay in native tokens OR ERC20
-     * @dev Allows to execute a transaction confirmed by required signature/s and then pays the account that submitted the transaction.
-     * @notice The fees are always transferred, even if the user transaction fails.
-     * @param _tx Smart Account transaction
-     * @param refundInfo Required information for gas refunds
-     * @param signatures Packed signature/s data ({bytes32 r}{bytes32 s}{uint8 v})
+     * Validate the nonce of the UserOperation.
+     * This method may validate the nonce requirement of this account.
+     * e.g.
+     * To limit the nonce to use sequenced UserOps only (no "out of order" UserOps):
+     *      `require(nonce < type(uint64).max)`
+     * For a hypothetical account that *requires* the nonce to be out-of-order:
+     *      `require(nonce & type(uint64).max == 0)`
+     *
+     * The actual nonce uniqueness is managed by the EntryPoint, and thus no other
+     * action is needed by the account itself.
+     *
+     * @param nonce to validate
+     *
+     * solhint-disable-next-line no-empty-blocks
      */
-    function execTransaction(
-        Transaction memory _tx,
-        FeeRefund memory refundInfo,
-        bytes memory signatures
-    ) external payable virtual returns (bool success);
+    function _validateNonce(uint256 nonce) internal view virtual {
+        (nonce);
+    }
```

Comment:
this internal function can be removed since it's not used

Reply:
if we remove in abstract class then no need here.

if maintained abstract in abstract class then yes we need to write some implementation which would be this.

Reply:
another idea would be to keep it implemented yet virtual in abstract base class, and not mention it at all in implementation contracts

Reply:
yeah can implement the default implementation in abstract class with a revert statement with message, "This method is not implemented"

Reply:
why don't just leave it empty?

Reply:
Our codebase is getting bigger we should try to keep it as simple as possible. For "internal" functions that are not used we should remove them, it won't change our ABI and probably not the bytecode as well. While 100% agree that public/external functions might be good to keep even if empty, internals should be removed IMO. We can always reintroduce them if necessary.

Reply:
removed